File: defaults-edit/FastHexTransformer.m
//
//  FastHexTransformer.m
//  defaults-edit
//
//  Created by Ian Gregory on 23-03-2019.
//  Copyright © 2019 Ian Gregory. All rights reserved.
//

#import "FastHexTransformer.h"
#import "NSData+FastHex.h"

// TODO: Convert to an NSFormatter

@implementation FastHexTransformer

- (NSString *)transformedValue:(id)value {
    if ([value isKindOfClass:[NSData class]]) {
        return [(NSData *)value hexStringRepresentationUppercase:NO];
    } else {
        return @"";
    }
}

- (NSData *)reverseTransformedValue:(NSString *)value {
    return [[NSData alloc] initWithHexString:value ignoreOtherCharacters:YES];
}

@end

File: defaults edit/PlistEditor/FastHexTransformer.swift
//
//  FastHexTransformer.swift
//  defaults edit
//
//  Created by Ian Gregory on 6 Jun ’20.
//  Copyright © 2020 Ian Gregory. All rights reserved.
//

import Foundation
import Data_FastHex

// TODO: Convert to an NSFormatter?

@objc(FastHexTransformer)
class FastHexTransformer: ValueTransformer {
    
    override func transformedValue(_ value: Any?) -> Any? {
        (value as? NSData)?.lowercaseHexStringRepresentation ?? ""
    }
    
    override func reverseTransformedValue(_ value: Any?) -> Any? {
        NSData(hexString: value as! String)
    }
    
}
